nsds_lab_to_nwb/components/stimulus/stim_value_extractor.py

```diff
@@ -2,67 +2,116 @@
+
+        raise ValueError('Unknown input for stimulus parameterization')
+
+    def _get_stim_values_path(self, path_from_metadata=None):
+        # prioritize path from list_of_stimuli.yaml in this package
+        path_from_los = self.stim_info['parameter_path']
+        if len(path_from_los) > 0:
+            return os.path.join(self.stim_lib_path, path_from_los)
+
+        # if empty path in list_of_stimuli.yaml, use metadata input
+        return os.path.join(self.stim_lib_path, path_from_metadata)
 
 
 def tone_stimulus_values(mat_file_path):
-    ''' adapted from mars.configs.block_directory '''
-    with h5py.File(mat_file_path, 'r') as sio:
-        stim_vals = sio['stimVls'][:].astype(int)
-    stim_vals[0,:] = stim_vals[0,:]+8
+    ''' adapted from mars.configs.block_directory
+
+    Parameters:
+    -----------
+    mat_file_path: full path to a .mat file that contains stim_values.
+
+    Returns:
+    --------
+    stim_vals: a 2D array with two rows along the leading dimension.
+        stim_vals[0, :] are the amplitudes,
+        stim_vals[1, :] are the frequencies of the tones.
+        (should confirm!)
+    '''
+    sio = read_mat_file(mat_file_path)
+    stim_vals = sio['stimVls'][:].astype(int)
+
+    # check dimension
+    shape = stim_vals.shape
+    if not (len(shape) == 2) and (2 in shape):
+        # should be a 2D array, with 2 rows (or 2 columns)
+        raise ValueError('stim_vals dimension mismatch')
+    if shape[1] == 2:
+        stim_vals = stim_vals.T
+
+    # this offset value comes from mars; what is this?
+    # variable naming (amp_offset) was by JHB and could be wrong
+    amp_offset = 8
+    stim_vals[0,:] = stim_vals[0,:] + amp_offset
+
     return stim_vals
 
-def timit_stimulus_values(csv_file_path):
-    ''' adapted from mars.configs.block_directory '''
-    # NOTE: this file timit998.txt is just a list of wav files.
-    # TODO: also load from these wav files.
-    stim_vals = []
-    with open(csv_file_path, 'r') as f:
-        reader = csv.DictReader(f)
-        for row in reader:
-            stim_vals.append(row['sample_id'])
+
+def timit_stimulus_values(file_path):
+    ''' adapted from mars.configs.block_directory
+
+    Parameters:
+    -----------
+    file_path: full path to a .txt file that contains a list of filenames
+
+    Returns:
+    --------
+    stim_vals: a list of strings, where each item is a .wav file name in TIMIT.
+        (should confirm!)
+    '''
+    _, ext = os.path.splitext(file_path)
+    if not (ext == '.txt'):
+        raise ValueError('for now only accepting a txt file that lists wav file names.')
+
+    # expecting a text file, one .wav filename string per row
+    with open(file_path) as f:
+        stim_vals = f.readlines()
     return stim_vals
 
+
 def gen_tone_stim_vals():
-    ''' exact copy from mars.configs.block_directory '''
+    ''' exact copy from mars.configs.block_directory
+
+    Note: this is only used for test_tone_stim.yaml (see metadata repo)
```

Comment:
If we ever actually use this, I think we should just write the values out to a file (not in this PR). I'll make an issue.